### ISSUE TITLE
typeof function guard doesn't work after update

### ISSUE THREAD

**Issue by orlov-vo** [Microsoft/TypeScript#28549]
**TypeScript Version:**  3.1.6 (previous: 3.0.1)

**Code**

```ts
const data = typeof fetcherParams === 'function'
  ? fetcherParams()
  : fetcherParams
```

It doesn't work for me. Also, I tried to use `instanceof` and it works:

```ts
const data = fetcherParams instanceof Function
  ? fetcherParams()
  : fetcherParams
```

**Expected behavior:** no error with `typeof` statement

**Actual behavior:**

```
TS2349: Cannot invoke an expression whose type lacks a call signature. Type 'Function | (() => object)' has no compatible call signatures.
```

**Comment by jack-williams** [comment]
I think this is working as intended after a (breaking) change in 3.1.

Read [here](https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#narrowing-functions-now-intersects--object-and-unconstrained-generic-type-parameters).
